--- build/config.rb.origo	2012-08-22 20:22:08.988860031 +0200
+++ build/config.rb	2012-08-22 20:30:00.184849959 +0200
@@ -40,7 +40,7 @@
 
 # Extra linker flags that should always be passed to the linker.
 # Should be included last in the command string, even after PlatformInfo.portability_ldflags.
-EXTRA_LDFLAGS  = ""
+EXTRA_LDFLAGS  = "#{ENV['LDFLAGS']}"
 
 # Whether to use the vendored libev or the system one.
-USE_VENDORED_LIBEV = boolean_option("USE_VENDORED_LIBEV", true)
+USE_VENDORED_LIBEV = false
